packages/apollo-server-express/src/ApolloServer.ts

```diff
@@ -142,12 +169,49 @@ export class ApolloServer extends ApolloServerBase {
     // schema, you'll need to manually specify `introspection: true` in the
     // ApolloServer constructor; by default, the introspection query is only
     // enabled in dev.
-    const guiEnabled =
-      !!gui || (gui === undefined && process.env.NODE_ENV !== 'production');
-
     app.use(path, (req, res, next) => {
-      if (guiEnabled && req.method === 'GET') {
+      // We cannot do an existence check on `gui` since it may be a boolean
+      const nonDynamicGui: NonDynamicGuiConfig =
```

Comment:
Can we put this configuration in a helper function and place the setting of `guiOptions` after the cors options above?

It will put the configuration code together and  reduce the complexity of this middleware